x-pack/plugins/task_manager/server/saved_objects/migrations.ts

```diff
@@ -5,16 +5,119 @@
  * 2.0.
  */
 
-import { SavedObjectMigrationMap, SavedObjectUnsanitizedDoc } from '../../../../../src/core/server';
+import {
+  LogMeta,
+  SavedObjectMigrationContext,
+  SavedObjectMigrationFn,
+  SavedObjectMigrationMap,
+  SavedObjectsUtils,
+  SavedObjectUnsanitizedDoc,
+} from '../../../../../src/core/server';
 import { TaskInstance, TaskInstanceWithDeprecatedFields } from '../task';
 
-export const migrations: SavedObjectMigrationMap = {
-  '7.4.0': (doc) => ({
-    ...doc,
-    updated_at: new Date().toISOString(),
-  }),
-  '7.6.0': moveIntervalIntoSchedule,
-};
+interface TaskInstanceLogMeta extends LogMeta {
+  migrations: { taskInstanceDocument: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields> };
+}
+
+type TaskInstanceMigration = (
+  doc: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>
+) => SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>;
+
+export function getMigrations(): SavedObjectMigrationMap {
+  return {
+    '7.4.0': executeMigrationWithErrorHandling(
+      (doc) => ({
+        ...doc,
+        updated_at: new Date().toISOString(),
+      }),
+      '7.4.0'
+    ),
+    '7.6.0': executeMigrationWithErrorHandling(moveIntervalIntoSchedule, '7.16.0'),
+    '8.0.0': executeMigrationWithErrorHandling(
+      pipeMigrations(alertingTaskLegacyIdToSavedObjectIds, actionsTasksLegacyIdToSavedObjectIds),
+      '8.0.0'
+    ),
+  };
+}
+
+function executeMigrationWithErrorHandling(
+  migrationFunc: SavedObjectMigrationFn<
+    TaskInstanceWithDeprecatedFields,
+    TaskInstanceWithDeprecatedFields
+  >,
+  version: string
+) {
+  return (
+    doc: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>,
+    context: SavedObjectMigrationContext
+  ) => {
+    try {
+      return migrationFunc(doc, context);
+    } catch (ex) {
+      context.log.error<TaskInstanceLogMeta>(
+        `savedObject ${version} migration failed for task instance ${doc.id} with error: ${ex.message}`,
+        {
+          migrations: {
+            taskInstanceDocument: doc,
+          },
+        }
+      );
+      throw ex;
+    }
+  };
+}
+
+function alertingTaskLegacyIdToSavedObjectIds(
+  doc: SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields>
+): SavedObjectUnsanitizedDoc<TaskInstanceWithDeprecatedFields> {
+  if (doc.attributes.taskType.startsWith('alerting:')) {
+    const params = doc.attributes.params;
```

Comment:
`params` are stored as stringified JSON, so I I think you need a JSON.parse before trying to access the `alertId` and `spaceId`

Example:

```
"params" : """{"alertId":"47ac87c0-fb8a-11eb-a421-c77e100f1b72","spaceId":"new-space"}""",
```

Reply:
Thank you, @ymao1. I was wonder because the type of params is `'Record<string, any>' `. I will fix that.